cache/in-memory/src/event/member.rs

```diff
@@ -136,6 +136,10 @@ impl UpdateCache for MemberAdd {
             .entry(self.guild_id)
             .or_default()
             .insert(self.0.user.id);
+
```

Comment:
Move this above the `RecourceType::MEMBER` check and check whether the user wants to cache guilds `ResourceType::GUILD`

Reply:
done